### PR TITLE
feat: Hermes cli

### PR DESCRIPTION
**Pull request by Mr-Leshiy**
# Description

Added a cli functionality for hermes with a simple `run` command which loads and runs a HermesApp

### CODE REVIEW

**Review by minikin — CHANGES_REQUESTED** [review]
Let's add/update README on how to use it.

**Comment by stevenj** [comment]
This PR conflates multiple things.

1. Why did you remove Build Info which was specifically added at my request?  I wan't all our applications to report how they are built, version etc.  So why was it removed?
2. Why have we refactored all of the logging?  We should not be just changing behaviour without it being discussed.  Logging was done how it was because its a step towards a proper logging solution, this change does not advance that solution.  We should not be doing work on issues that are not issues.  The proper solution is to add a subscriber for all log messages, which categorises them and stores them for reporting through a future UI OR sends them to stdout/stderr.  These changes don't go in that direction that I can see and so are wasdted effort.  This is why we should not be adding things which are not scoped and discussed.
3. Why do we need a "run" cli command?  Hermes should just run if its not given any other option, I also want to NOT add tons of CLI arguments and options, we have already discussed this.  When we allow hermes to do other things, they can be commands with options, but running the hermes executable itself should never need "run".

So, I wonder why any of this was done?  

At the very least these three changes should be in their own PR.  But that said, changes 1 and 2 are not necessary and revert work already done intentionally, so should be part of any PR.

**Review by stevenj — CHANGES_REQUESTED** [review]
See comment i left

File: hermes/crates/cbork/cddl-parser/Cargo.toml [hunk 1]
```diff
@@ -12,7 +12,6 @@ license = "MIT OR Apache-2.0"
 workspace = true
 
 [dependencies]
-derive_more =  "0.99.17"
```
[the remark this answers is not in the document]
Comment (Mr-Leshiy) [reply hermes/crates/cbork/cddl-parser/Cargo.toml:15]: I didn't change it that way, they are in their own workspace.

**Review by stevenj — CHANGES_REQUESTED** [review]

**Review by stevenj — APPROVED** [review]

**Review by minikin — APPROVED** [review]

**Pull request merged**